src/Gateway.php

```diff
@@ -116,63 +114,118 @@ public function payment_fields() {
-		$fields['billing_customer_reference1'] = array(
-			'label'       => __( 'Reference 1', 'ledyer-payments-for-woocommerce' ),
-			'placeholder' => __( 'For example, reference code or cost center', 'ledyer-payments-for-woocommerce' ),
-			'required'    => false,
-			'class'       => array( 'form-row-wide' ),
-			'clear'       => true,
-			'priority'    => 35,
+	public function additional_admin_billing_fields( $billing_fields, $order, $context = 'edit' ) {
+		if ( ! $this->is_chosen_gateway( $order ) ) {
+			return $billing_fields;
+		}
+
+		// The value can be explicitly set by adding a 'value' property. However, if not set, WC will attempt to prefix the field name with 'billing_', and retrieve it from a metadata field by that key. E.g., 'company_number' => $order->get_meta('_billing_company_number').
+		$additional_fields = array(
+			'company_number'              => array(
+				'label' => __( 'Company number', 'ledyer-payments-for-woocommerce' ),
+				'type'  => 'text',
+				'show'  => true,
+			),
+			'ledyer_customer_reference_1' => array(
+				'id'    => '_ledyer_customer_reference_1',
+				'label' => __( 'Customer reference 1', 'ledyer-payments-for-woocommerce' ),
+				'type'  => 'text',
+				'show'  => true,
+				'value' => $order->get_meta( '_ledyer_customer_reference_1' ),
+			),
+			'ledyer_customer_reference_2' => array(
+				'id'    => '_ledyer_customer_reference_2',
+				'label' => __( 'Customer reference 2', 'ledyer-payments-for-woocommerce' ),
+				'type'  => 'text',
+				'show'  => true,
+				'value' => $order->get_meta( '_ledyer_customer_reference_2' ),
+			),
 		);
 
-		$fields['billing_customer_reference2'] = array(
-			'label'       => __( 'Reference 2', 'ledyer-payments-for-woocommerce' ),
-			'placeholder' => __( 'For example, purchase order number', 'ledyer-payments-for-woocommerce' ),
-			'required'    => false,
-			'class'       => array( 'form-row-wide' ),
-			'clear'       => true,
-			'priority'    => 35,
+		// Inserts the additional fields after the 'company' field.
+		$index = array_search(
+			'company',
+			array_keys( $billing_fields ),
+			true
 		);
 
-		return $fields;
+		// Increment $index by 1 to include the first element too (since $index starts from 0, but length starts from 1).
+		$before = array_slice( $billing_fields, 0, $index + 1 );
+		$after  = array_slice( $billing_fields, $index );
+		return $before + $additional_fields + $after;
 	}
 
 	/**
-	 * Add additional billing fields to the admin order page.
+	 * Process the admin order update.
 	 *
-	 * @param array $fields The billing fields.
-	 * @return array
+	 * @param int $order_id The WooCommerce order id.
+	 * @return void
 	 */
-	public function additional_admin_billing_fields( $fields ) {
-		$fields['customer_reference1'] = array(
-			'label' => __( 'Customer reference 1', 'ledyer-payments-for-woocommerce' ),
-			'show'  => true,
-		);
+	public function admin_order_processing( $order_id ) {
```

Comment:
Change the name of this method to match what it does. for example `update_additional_admin_billing_fields` or something similar. That makes it easier when looking at things like hooks as well to see what we are actually doing without reading the code in the method.